sklearn/neighbors/regression.py

```diff
@@ -148,7 +148,7 @@ def predict(self, X):
             Target values
         """
         if issparse(X) and self.metric == 'precomputed':
-            raise ValueError(
+            raise TypeError(
```

Comment:
@agramfort should  I leave this for another PR or is it ok to change it here ?

Reply:
@agramfort related to our discussion, in fact here they do call `check_array` but the sparse error is raised by this "if", which has a `ValueError`, so I guess this is the right fix (and I'll let it in this PR for this one as you said)